lib/reverse_sentence.rb

```diff
@@ -1,6 +1,52 @@
-# A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# A method to reverse the words in a sentence, in place
+
+# Time complexity: O(n) - my reverse_from_a_to_b method does not depend on the length of my_sentence,
+# the only method that is invoked and depends on the length of my_sentence is reverse_each_word_of_sentence
+
+# Space complexity: O(1) - the length of my_sentence does does not affect the space taken up to save our variable;
+# all strings take up the same space regardless of length; unlike arrays
+
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+
+  class << self
+    def reverse_from_a_to_b(str, a, b)
+      while a < b
+        temp = str[a]
+        str[a] = str[b]
+        str[b] = temp
+        a += 1
+        b -= 1
+      end
+      return str
+    end
+  end
+
+  class << self
```

Comment:
I don't think you need this `class << self` here.  Where did you get this from?

Reply:
Hey Chris! I was getting the following error "method definitions must not be nested. use lambda instead", so I googled it and found this solution "# good

def foo
  class << self
    def bar
    end
  end
end" on [https://msp-greg.github.io/rubocop/RuboCop/Cop/Lint/NestedMethodDefinition.html](url)